Build failure: heimdal

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix-build -A heimdal`

### Build log

```
Making all in hdb
make[2]: Entering directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-2/source/lib/hdb'
../../lib/asn1/asn1_compile --option-file=./hdb.opt ./hdb.asn1 hdb_asn1
for genfile in 'asn1_Event.c asn1_GENERATION.c asn1_HDB_EncTypeList.c asn1_HDB_Ext_Aliases.c asn1_HDB_Ext_Constrained_delegation_acl.c asn1_HDB_Ext_KeyRotation.c asn1_HDB_Ext_KeySet.c asn1_HDB_Ext_Lan_Manager_
OWF.c asn1_HDB_Ext_Password.c asn1_HDB_Ext_PKINIT_acl.c asn1_HDB_Ext_PKINIT_cert.c asn1_HDB_Ext_PKINIT_hash.c asn1_HDB_EntryOrAlias.c asn1_HDB_entry_alias.c asn1_HDB_entry.c asn1_HDB_extension.c asn1_HDB_exten
sions.c asn1_HDB_keyset.c asn1_HDBFlags.c asn1_Key.c asn1_KeyRotation.c asn1_KeyRotationFlags.c asn1_Keys.c asn1_Salt.c'; do \
        clang-format -style='{BasedOnStyle: Mozilla, AlwaysBreakAfterReturnType: TopLevelDefinitions, IndentWidth: 4, SortIncludes: false}' -i ${genfile}; \
done
../../lib/com_err/compile_et hdb_err.et
cd .; perl ../../cf/make-proto.pl -q -P comment -o hdb-protos.h common.c db.c db3.c ext.c hdb-ldap.c hdb.c hdb-sqlite.c hdb-keytab.c hdb-mdb.c hdb-mitdb.c hdb_locl.h keys.c keytab.c dbinfo.c mkey.c ndbm.c prin
t.c || rm -f hdb-protos.h
updating hdb-protos.h
cd .; perl ../../cf/make-proto.pl -q -P comment -p hdb-private.h common.c db.c db3.c ext.c hdb-ldap.c hdb.c hdb-sqlite.c hdb-keytab.c hdb-mdb.c hdb-mitdb.c hdb_locl.h keys.c keytab.c dbinfo.c mkey.c ndbm.c pri
nt.c || rm -f hdb-private.h
updating hdb-private.h
make  all-am
make[3]: Entering directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-2/source/lib/hdb'
  CC       test_dbinfo.o
  CC       common.lo
  CC       db.lo
db.c:54:5: error: use of undeclared identifier 'DB'
    DB *d = (DB*)db->hdb_db;
    ^
db.c:54:9: error: use of undeclared identifier 'd'
    DB *d = (DB*)db->hdb_db;
        ^
db.c:54:17: error: expected expression
    DB *d = (DB*)db->hdb_db;
                ^
db.c:54:14: error: use of undeclared identifier 'DB'
    DB *d = (DB*)db->hdb_db;
             ^
db.c:56:17: error: use of undeclared identifier 'd'
    heim_assert(d != 0, "Closing already closed HDB");
                ^
db.c:58:7: error: use of undeclared identifier 'd'
    (*d->close)(d);
      ^
db.c:58:17: error: use of undeclared identifier 'd'
    (*d->close)(d);
                ^
db.c:85:5: error: use of undeclared identifier 'DB'
    DB *d = (DB*)db->hdb_db;
    ^
db.c:85:9: error: use of undeclared identifier 'd'
    DB *d = (DB*)db->hdb_db;
        ^
db.c:85:17: error: expected expression
    DB *d = (DB*)db->hdb_db;
                ^
db.c:85:14: error: use of undeclared identifier 'DB'
    DB *d = (DB*)db->hdb_db;
             ^
db.c:90:17: error: use of undeclared identifier 'd'
        ret = (*d->sync)(d, 0);
                ^
db.c:90:26: error: use of undeclared identifier 'd'
        ret = (*d->sync)(d, 0);
                         ^
db.c:119:5: error: use of undeclared identifier 'DB'
    DB *d = (DB*)db->hdb_db;
    ^
db.c:119:9: error: use of undeclared identifier 'd'
    DB *d = (DB*)db->hdb_db;
        ^
db.c:119:17: error: expected expression
    DB *d = (DB*)db->hdb_db;
                ^
db.c:119:14: error: use of undeclared identifier 'DB'
    DB *d = (DB*)db->hdb_db;
             ^
db.c:120:5: error: use of undeclared identifier 'DBT'
    DBT key, value;
    ^
db.c:124:14: error: use of undeclared identifier 'd'
    code = (*d->seq)(d, &key, &value, flag);
             ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[3]: *** [Makefile:1115: db.lo] Error 1
make[3]: Leaving directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-2/source/lib/hdb'
make[2]: *** [Makefile:932: all] Error 2
make[2]: Leaving directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-2/source/lib/hdb'
make[1]: *** [Makefile:590: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-2/source/lib'
make: *** [Makefile:634: all-recursive] Error 1
```

### Additional context

This is happening on Darwin. The failure was triggered by https://github.com/NixOS/nixpkgs/pull/345129 that bumped the version.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@jsierles @SuperSandro2000 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.6.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/ywvkxmphiy3d110p70qj174ici3zmhwg-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

More observations.

1. If I remove berkeley-db arguments to configure script (and the `db` dependency), then compilation passes. But I'm not sure it's a good way to resolve the problem. (AFAIU Berkeley is [advised](https://github.com/heimdal/heimdal/wiki/Building-and-installing#building) by upstream?)

---

2. The reason why `lib/hdb/db.c` fails to compile is because it doesn't include a file with its definitions. Manually defining `HAVE_DB_185_H` in `config.h` after `configurePhase` makes the library build. (There seems to be no way to define this macro except by modifying `config.h`...)

---

But...

3. It then fails on `lib/otp` as follows:

```
make[2]: Entering directory '/private/tmp/tmp.LIUy4MZguk/source/lib/otp'
  CC       otp_db.lo
clang version 16.0.6
Target: arm64-apple-darwin
Thread model: posix
InstalledDir: /nix/store/arqjq1nskg65clyh1zy8fm21jr7cypq2-clang-16.0.6/bin
 (in-process)
 "/nix/store/arqjq1nskg65clyh1zy8fm21jr7cypq2-clang-16.0.6/bin/clang-16" -cc1 -triple arm64-apple-macosx11.0.0 -Wundef-prefix=TARGET_OS_ -Werror=undef-prefix -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -emit-obj -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name otp_db.c -mrelocation-model pic -pic-level 2 -mframe-pointer=non-leaf -ffp-contract=on -fno-rounding-math -funwind-tables=1 -target-sdk-version=11.0 -fcompatibility-qualified-id-block-type-checking -fvisibility-inlines-hidden-static-local-var -target-cpu apple-m1 -target-feature +v8.5a -target-feature +crc -target-feature +crypto -target-feature +dotprod -target-feature +fp-armv8 -target-feature +fp16fml -target-feature +lse -target-feature +ras -target-feature +rcpc -target-feature +rdm -target-feature +neon -target-feature +zcm -target-feature +zcz -target-feature +fullfp16 -target-feature +sm4 -target-feature +sha3 -target-feature +sha2 -target-feature +aes -target-abi darwinpcs -mllvm -treat-scalable-fixed-error-as-warning -debug-info-kind=standalone -dwarf-version=4 -debugger-tuning=lldb -target-linker-version 951.9 -v -fcoverage-compilation-dir=/tmp/tmp.LIUy4MZguk/source/lib/otp -nostdsysteminc -resource-dir /nix/store/22a6p8311lg3z4ksrjbj018i7z3vgkva-clang-wrapper-16.0.6/resource-root -dependency-file .deps/otp_db.Tpo -MT otp_db.lo -sys-header-deps -MP -idirafter /nix/store/kk9lnicm474vif1bmhf1268xzrn7z35f-libSystem-11.0.0/include -isysroot /nix/store/34dfpd1mff77lnv9yq7z2f0i2xzgw4iv-sdkroot-11.0/MacOSX11.0.sdk -isystem /nix/store/lg1kmmrq2m5f6bv5lsdm222mldjchw9b-gettext-0.21.1/include -isystem /nix/store/a4pvlm827kdpji89y3s4kay1l82w6q53-libtool-2.4.7/include -isystem /nix/store/hkz3rjjs88q1dhb1s0hxwwzj4i122nj6-file-5.45-dev/include -isystem /nix/store/ybnf7k6i9p244bbhsbxizqk65z58cwyr-python3-3.12.6/include -isystem /nix/store/9clz7185w6gf2nrkdhywfaib44s6wzfa-libxcrypt-4.4.36/include -isystem /nix/store/c242dkwj4w2dfzajr6anh0ynyyjfs21z-flex-2.6.4/include -isystem /nix/store/vcj58ndz7fxjpsl40g646nwnrg3g5h4w-curl-8.9.1-dev/include -isystem /nix/store/9cyj0710q16v51xwdn6cb84hvfm4j8ys-brotli-1.1.0-dev/include -isystem /nix/store/7bbjc5z5ach2gn1gd5x9qjqkrv56azlp-krb5-1.21.3-dev/include -isystem /nix/store/3r5kxw0ihas9b6z7l48b1if3wyq4i1ni-apple-lib-xpc/include -iframework /nix/store/w1ypff08bd7kv4r4rkw3yyr0pmynvlk3-apple-framework-Kerberos-11.0.0/Library/Frameworks -isystem /nix/store/g3c2zbnsxfa5rwl3xq30qvsfnlh0njhj-nghttp2-1.63.0-dev/include -isystem /nix/store/j69mbrlrqzszrivl3mg9qh555xwiiv8h-libidn2-2.3.7-dev/include -isystem /nix/store/0w7wifk6j8df14dh887xk7zrjgv9b86a-openssl-3.3.2-dev/include -isystem /nix/store/6pz2rg05irlfxwxxdl2b9f0gl6b4jw58-libpsl-0.21.5-dev/include -isystem /nix/store/dmisjm8ivz5aabms4ijxr03m44hl5zhk-libssh2-1.11.0-dev/include -isystem /nix/store/7wn6iqdw56zwly413gk0zsyhzzh0zjcd-zlib-1.3.1-dev/include -isystem /nix/store/8pp9xsgpmhsxn7b3j6sanc8h6gqvfldm-zstd-1.5.6-dev/include -iframework /nix/store/nhf361hkyv2p2cy7nx587sf936vzpvcm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks -isystem /nix/store/3cmjqi371abl1amwrm5v653ls2cq3kc0-libobjc-11.0.0/include -iframework /nix/store/cql794qjgam02kwliv005vwlgh97bf44-apple-framework-CoreServices-11.0.0/Library/Frameworks -iframework /nix/store/2azv2z23zwcr5arx69g8zy0c0r6l51f2-apple-framework-CFNetwork-11.0.0/Library/Frameworks -iframework /nix/store/62lyxlwf72lsf5jq68ia537ypyllqnfb-apple-framework-CoreAudio-11.0.0/Library/Frameworks -iframework /nix/store/spfayx4rjhbamkxcdh790qfis3y1dha2-apple-framework-CoreAudioTypes-11.0.0/Library/Frameworks -iframework /nix/store/y38j642asaqci7mry5hz0yfyrga1ahn4-apple-framework-IOKit-11.0.0/Library/Frameworks -iframework /nix/store/1r9344446jjd34kr7wl8ijjnswir4vmc-apple-framework-DiskArbitration-11.0.0/Library/Frameworks -iframework /nix/store/9i9k3xqbb6j4w7ir6hbn66ddyh3bsl2a-apple-framework-NetFS-11.0.0/Library/Frameworks -iframework /nix/store/mf7qx60b8hc6mvpm242ldw09jhpy5xg5-apple-framework-Security-11.0.0/Library/Frameworks -isystem /nix/store/4pkqd54rlisrikgkfsr42nxh9kdsma9p-apple-lib-libDER/include -iframework /nix/store/i90055z39g08n2jip1zvagbrvc0s7n49-apple-framework-ServiceManagement-11.0.0/Library/Frameworks -iframework /nix/store/lc82lgkvqlvy11sgkr9ci2fp25pymq2r-apple-framework-SystemConfiguration-11.0.0/Library/Frameworks -isystem /nix/store/0q256qa09wnplr1fs9c24fm59mn8l76n-zulu-ca-jdk-21.0.2/include -isystem /nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include -isystem /nix/store/fv74pj6wlk9gpapc4n7z09z3kl7h07ry-compiler-rt-libc-16.0.6-dev/include -isystem /nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include -isystem /nix/store/1fm6rjrs9q4dj10fdiggbfjfa2svgbq2-libedit-20240808-3.1-dev/include -isystem /nix/store/1g7lxsx2plfs828hr8y73xhjikzvn4b8-ncurses-6.4.20221231-dev/include -isystem /nix/store/2mrhx9jihzhyz5bi853dmxsxmzrszwbk-openpam-20230627/include -isystem /nix/store/nafzxyi3z9bib531s371wfjv1x08c2z5-cjson-1.7.18/include -isystem /nix/store/mlc8hi0b9r4ik64g81gppjw4r5i6692a-openldap-2.6.8-dev/include -isystem /nix/store/x5kfhipq741nbcz36i397biw3hpkkfnx-sqlite-3.46.1-dev/include -iframework /nix/store/82m40ksmhhjhcmbfc8n8vip58wbq8cxm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks -isystem /nix/store/lg1kmmrq2m5f6bv5lsdm222mldjchw9b-gettext-0.21.1/include -isystem /nix/store/a4pvlm827kdpji89y3s4kay1l82w6q53-libtool-2.4.7/include -isystem /nix/store/hkz3rjjs88q1dhb1s0hxwwzj4i122nj6-file-5.45-dev/include -isystem /nix/store/ybnf7k6i9p244bbhsbxizqk65z58cwyr-python3-3.12.6/include -isystem /nix/store/9clz7185w6gf2nrkdhywfaib44s6wzfa-libxcrypt-4.4.36/include -isystem /nix/store/c242dkwj4w2dfzajr6anh0ynyyjfs21z-flex-2.6.4/include -isystem /nix/store/vcj58ndz7fxjpsl40g646nwnrg3g5h4w-curl-8.9.1-dev/include -isystem /nix/store/9cyj0710q16v51xwdn6cb84hvfm4j8ys-brotli-1.1.0-dev/include -isystem /nix/store/7bbjc5z5ach2gn1gd5x9qjqkrv56azlp-krb5-1.21.3-dev/include -isystem /nix/store/3r5kxw0ihas9b6z7l48b1if3wyq4i1ni-apple-lib-xpc/include -iframework /nix/store/w1ypff08bd7kv4r4rkw3yyr0pmynvlk3-apple-framework-Kerberos-11.0.0/Library/Frameworks -isystem /nix/store/g3c2zbnsxfa5rwl3xq30qvsfnlh0njhj-nghttp2-1.63.0-dev/include -isystem /nix/store/j69mbrlrqzszrivl3mg9qh555xwiiv8h-libidn2-2.3.7-dev/include -isystem /nix/store/0w7wifk6j8df14dh887xk7zrjgv9b86a-openssl-3.3.2-dev/include -isystem /nix/store/6pz2rg05irlfxwxxdl2b9f0gl6b4jw58-libpsl-0.21.5-dev/include -isystem /nix/store/dmisjm8ivz5aabms4ijxr03m44hl5zhk-libssh2-1.11.0-dev/include -isystem /nix/store/7wn6iqdw56zwly413gk0zsyhzzh0zjcd-zlib-1.3.1-dev/include -isystem /nix/store/8pp9xsgpmhsxn7b3j6sanc8h6gqvfldm-zstd-1.5.6-dev/include -iframework /nix/store/nhf361hkyv2p2cy7nx587sf936vzpvcm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks -isystem /nix/store/3cmjqi371abl1amwrm5v653ls2cq3kc0-libobjc-11.0.0/include -iframework /nix/store/cql794qjgam02kwliv005vwlgh97bf44-apple-framework-CoreServices-11.0.0/Library/Frameworks -iframework /nix/store/2azv2z23zwcr5arx69g8zy0c0r6l51f2-apple-framework-CFNetwork-11.0.0/Library/Frameworks -iframework /nix/store/62lyxlwf72lsf5jq68ia537ypyllqnfb-apple-framework-CoreAudio-11.0.0/Library/Frameworks -iframework /nix/store/spfayx4rjhbamkxcdh790qfis3y1dha2-apple-framework-CoreAudioTypes-11.0.0/Library/Frameworks -iframework /nix/store/y38j642asaqci7mry5hz0yfyrga1ahn4-apple-framework-IOKit-11.0.0/Library/Frameworks -iframework /nix/store/1r9344446jjd34kr7wl8ijjnswir4vmc-apple-framework-DiskArbitration-11.0.0/Library/Frameworks -iframework /nix/store/9i9k3xqbb6j4w7ir6hbn66ddyh3bsl2a-apple-framework-NetFS-11.0.0/Library/Frameworks -iframework /nix/store/mf7qx60b8hc6mvpm242ldw09jhpy5xg5-apple-framework-Security-11.0.0/Library/Frameworks -isystem /nix/store/4pkqd54rlisrikgkfsr42nxh9kdsma9p-apple-lib-libDER/include -iframework /nix/store/i90055z39g08n2jip1zvagbrvc0s7n49-apple-framework-ServiceManagement-11.0.0/Library/Frameworks -iframework /nix/store/lc82lgkvqlvy11sgkr9ci2fp25pymq2r-apple-framework-SystemConfiguration-11.0.0/Library/Frameworks -isystem /nix/store/0q256qa09wnplr1fs9c24fm59mn8l76n-zulu-ca-jdk-21.0.2/include -isystem /nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include -isystem /nix/store/fv74pj6wlk9gpapc4n7z09z3kl7h07ry-compiler-rt-libc-16.0.6-dev/include -isystem /nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include -isystem /nix/store/1fm6rjrs9q4dj10fdiggbfjfa2svgbq2-libedit-20240808-3.1-dev/include -isystem /nix/store/1g7lxsx2plfs828hr8y73xhjikzvn4b8-ncurses-6.4.20221231-dev/include -isystem /nix/store/2mrhx9jihzhyz5bi853dmxsxmzrszwbk-openpam-20230627/include -isystem /nix/store/nafzxyi3z9bib531s371wfjv1x08c2z5-cjson-1.7.18/include -isystem /nix/store/mlc8hi0b9r4ik64g81gppjw4r5i6692a-openldap-2.6.8-dev/include -isystem /nix/store/x5kfhipq741nbcz36i397biw3hpkkfnx-sqlite-3.46.1-dev/include -iframework /nix/store/82m40ksmhhjhcmbfc8n8vip58wbq8cxm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks -U _FORTIFY_SOURCE -D HAVE_CONFIG_H -I . -I . -I ../../include -I ../../include -I ../../lib/roken -I ../../lib/roken -D ROKEN_RENAME -I /nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include -D _LARGE_FILES= -D INET6 -D PIC -U _FORTIFY_SOURCE -D _FORTIFY_SOURCE=2 -F/nix/store/nhf361hkyv2p2cy7nx587sf936vzpvcm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks -F/nix/store/nhf361hkyv2p2cy7nx587sf936vzpvcm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks -internal-isystem /nix/store/22a6p8311lg3z4ksrjbj018i7z3vgkva-clang-wrapper-16.0.6/resource-root/include -source-date-epoch 315532800 -O2 -Wformat -Wformat-security -Werror=format-security -Wall -Wextra -Wno-sign-compare -Wno-unused-parameter -Wmissing-prototypes -Wpointer-arith -Wbad-function-cast -Wmissing-declarations -Wnested-externs -Wshadow -fdebug-compilation-dir=/tmp/tmp.LIUy4MZguk/source/lib/otp -ferror-limit 19 -fzero-call-used-regs=used-gpr-fwrapv -stack-protector 2 -stack-protector-buffer-size 4 -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fmax-type-align=16 -fcolor-diagnostics -vectorize-loops -vectorize-slp -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o .libs/otp_db.o -x c otp_db.c
clang -cc1 version 16.0.6 based upon LLVM 16.0.6 default target aarch64-apple-darwin23.6.0
ignoring duplicate directory "."
ignoring duplicate directory "../../include"
ignoring duplicate directory "../../lib/roken"
ignoring duplicate directory "/nix/store/nhf361hkyv2p2cy7nx587sf936vzpvcm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include"
  as it is a non-system directory that duplicates a system directory
ignoring duplicate directory "/nix/store/lg1kmmrq2m5f6bv5lsdm222mldjchw9b-gettext-0.21.1/include"
ignoring duplicate directory "/nix/store/a4pvlm827kdpji89y3s4kay1l82w6q53-libtool-2.4.7/include"
ignoring duplicate directory "/nix/store/hkz3rjjs88q1dhb1s0hxwwzj4i122nj6-file-5.45-dev/include"
ignoring duplicate directory "/nix/store/ybnf7k6i9p244bbhsbxizqk65z58cwyr-python3-3.12.6/include"
ignoring duplicate directory "/nix/store/9clz7185w6gf2nrkdhywfaib44s6wzfa-libxcrypt-4.4.36/include"
ignoring duplicate directory "/nix/store/c242dkwj4w2dfzajr6anh0ynyyjfs21z-flex-2.6.4/include"
ignoring duplicate directory "/nix/store/vcj58ndz7fxjpsl40g646nwnrg3g5h4w-curl-8.9.1-dev/include"
ignoring duplicate directory "/nix/store/9cyj0710q16v51xwdn6cb84hvfm4j8ys-brotli-1.1.0-dev/include"
ignoring duplicate directory "/nix/store/7bbjc5z5ach2gn1gd5x9qjqkrv56azlp-krb5-1.21.3-dev/include"
ignoring duplicate directory "/nix/store/3r5kxw0ihas9b6z7l48b1if3wyq4i1ni-apple-lib-xpc/include"
ignoring duplicate directory "/nix/store/g3c2zbnsxfa5rwl3xq30qvsfnlh0njhj-nghttp2-1.63.0-dev/include"
ignoring duplicate directory "/nix/store/j69mbrlrqzszrivl3mg9qh555xwiiv8h-libidn2-2.3.7-dev/include"
ignoring duplicate directory "/nix/store/0w7wifk6j8df14dh887xk7zrjgv9b86a-openssl-3.3.2-dev/include"
ignoring duplicate directory "/nix/store/6pz2rg05irlfxwxxdl2b9f0gl6b4jw58-libpsl-0.21.5-dev/include"
ignoring duplicate directory "/nix/store/dmisjm8ivz5aabms4ijxr03m44hl5zhk-libssh2-1.11.0-dev/include"
ignoring duplicate directory "/nix/store/7wn6iqdw56zwly413gk0zsyhzzh0zjcd-zlib-1.3.1-dev/include"
ignoring duplicate directory "/nix/store/8pp9xsgpmhsxn7b3j6sanc8h6gqvfldm-zstd-1.5.6-dev/include"
ignoring duplicate directory "/nix/store/3cmjqi371abl1amwrm5v653ls2cq3kc0-libobjc-11.0.0/include"
ignoring duplicate directory "/nix/store/4pkqd54rlisrikgkfsr42nxh9kdsma9p-apple-lib-libDER/include"
ignoring duplicate directory "/nix/store/0q256qa09wnplr1fs9c24fm59mn8l76n-zulu-ca-jdk-21.0.2/include"
ignoring duplicate directory "/nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include"
ignoring duplicate directory "/nix/store/fv74pj6wlk9gpapc4n7z09z3kl7h07ry-compiler-rt-libc-16.0.6-dev/include"
ignoring duplicate directory "/nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include"
ignoring duplicate directory "/nix/store/1fm6rjrs9q4dj10fdiggbfjfa2svgbq2-libedit-20240808-3.1-dev/include"
ignoring duplicate directory "/nix/store/1g7lxsx2plfs828hr8y73xhjikzvn4b8-ncurses-6.4.20221231-dev/include"
ignoring duplicate directory "/nix/store/2mrhx9jihzhyz5bi853dmxsxmzrszwbk-openpam-20230627/include"
ignoring duplicate directory "/nix/store/nafzxyi3z9bib531s371wfjv1x08c2z5-cjson-1.7.18/include"
ignoring duplicate directory "/nix/store/mlc8hi0b9r4ik64g81gppjw4r5i6692a-openldap-2.6.8-dev/include"
ignoring duplicate directory "/nix/store/x5kfhipq741nbcz36i397biw3hpkkfnx-sqlite-3.46.1-dev/include"
ignoring duplicate directory "/nix/store/nhf361hkyv2p2cy7nx587sf936vzpvcm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks"
  as it is a non-system directory that duplicates a system directory
ignoring duplicate directory "/nix/store/w1ypff08bd7kv4r4rkw3yyr0pmynvlk3-apple-framework-Kerberos-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/nhf361hkyv2p2cy7nx587sf936vzpvcm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/cql794qjgam02kwliv005vwlgh97bf44-apple-framework-CoreServices-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/2azv2z23zwcr5arx69g8zy0c0r6l51f2-apple-framework-CFNetwork-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/62lyxlwf72lsf5jq68ia537ypyllqnfb-apple-framework-CoreAudio-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/spfayx4rjhbamkxcdh790qfis3y1dha2-apple-framework-CoreAudioTypes-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/y38j642asaqci7mry5hz0yfyrga1ahn4-apple-framework-IOKit-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/1r9344446jjd34kr7wl8ijjnswir4vmc-apple-framework-DiskArbitration-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/9i9k3xqbb6j4w7ir6hbn66ddyh3bsl2a-apple-framework-NetFS-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/mf7qx60b8hc6mvpm242ldw09jhpy5xg5-apple-framework-Security-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/i90055z39g08n2jip1zvagbrvc0s7n49-apple-framework-ServiceManagement-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/lc82lgkvqlvy11sgkr9ci2fp25pymq2r-apple-framework-SystemConfiguration-11.0.0/Library/Frameworks"
ignoring duplicate directory "/nix/store/82m40ksmhhjhcmbfc8n8vip58wbq8cxm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 .
 ../../include
 ../../lib/roken
 /nix/store/lg1kmmrq2m5f6bv5lsdm222mldjchw9b-gettext-0.21.1/include
 /nix/store/a4pvlm827kdpji89y3s4kay1l82w6q53-libtool-2.4.7/include
 /nix/store/hkz3rjjs88q1dhb1s0hxwwzj4i122nj6-file-5.45-dev/include
 /nix/store/ybnf7k6i9p244bbhsbxizqk65z58cwyr-python3-3.12.6/include
 /nix/store/9clz7185w6gf2nrkdhywfaib44s6wzfa-libxcrypt-4.4.36/include
 /nix/store/c242dkwj4w2dfzajr6anh0ynyyjfs21z-flex-2.6.4/include
 /nix/store/vcj58ndz7fxjpsl40g646nwnrg3g5h4w-curl-8.9.1-dev/include
 /nix/store/9cyj0710q16v51xwdn6cb84hvfm4j8ys-brotli-1.1.0-dev/include
 /nix/store/7bbjc5z5ach2gn1gd5x9qjqkrv56azlp-krb5-1.21.3-dev/include
 /nix/store/3r5kxw0ihas9b6z7l48b1if3wyq4i1ni-apple-lib-xpc/include
 /nix/store/g3c2zbnsxfa5rwl3xq30qvsfnlh0njhj-nghttp2-1.63.0-dev/include
 /nix/store/j69mbrlrqzszrivl3mg9qh555xwiiv8h-libidn2-2.3.7-dev/include
 /nix/store/0w7wifk6j8df14dh887xk7zrjgv9b86a-openssl-3.3.2-dev/include
 /nix/store/6pz2rg05irlfxwxxdl2b9f0gl6b4jw58-libpsl-0.21.5-dev/include
 /nix/store/dmisjm8ivz5aabms4ijxr03m44hl5zhk-libssh2-1.11.0-dev/include
 /nix/store/7wn6iqdw56zwly413gk0zsyhzzh0zjcd-zlib-1.3.1-dev/include
 /nix/store/8pp9xsgpmhsxn7b3j6sanc8h6gqvfldm-zstd-1.5.6-dev/include
 /nix/store/3cmjqi371abl1amwrm5v653ls2cq3kc0-libobjc-11.0.0/include
 /nix/store/4pkqd54rlisrikgkfsr42nxh9kdsma9p-apple-lib-libDER/include
 /nix/store/0q256qa09wnplr1fs9c24fm59mn8l76n-zulu-ca-jdk-21.0.2/include
 /nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include
 /nix/store/fv74pj6wlk9gpapc4n7z09z3kl7h07ry-compiler-rt-libc-16.0.6-dev/include
 /nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include
 /nix/store/1fm6rjrs9q4dj10fdiggbfjfa2svgbq2-libedit-20240808-3.1-dev/include
 /nix/store/1g7lxsx2plfs828hr8y73xhjikzvn4b8-ncurses-6.4.20221231-dev/include
 /nix/store/2mrhx9jihzhyz5bi853dmxsxmzrszwbk-openpam-20230627/include
 /nix/store/nafzxyi3z9bib531s371wfjv1x08c2z5-cjson-1.7.18/include
 /nix/store/mlc8hi0b9r4ik64g81gppjw4r5i6692a-openldap-2.6.8-dev/include
 /nix/store/x5kfhipq741nbcz36i397biw3hpkkfnx-sqlite-3.46.1-dev/include
 /nix/store/w1ypff08bd7kv4r4rkw3yyr0pmynvlk3-apple-framework-Kerberos-11.0.0/Library/Frameworks (framework directory)
 /nix/store/nhf361hkyv2p2cy7nx587sf936vzpvcm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks (framework directory)
 /nix/store/cql794qjgam02kwliv005vwlgh97bf44-apple-framework-CoreServices-11.0.0/Library/Frameworks (framework directory)
 /nix/store/2azv2z23zwcr5arx69g8zy0c0r6l51f2-apple-framework-CFNetwork-11.0.0/Library/Frameworks (framework directory)
 /nix/store/62lyxlwf72lsf5jq68ia537ypyllqnfb-apple-framework-CoreAudio-11.0.0/Library/Frameworks (framework directory)
 /nix/store/spfayx4rjhbamkxcdh790qfis3y1dha2-apple-framework-CoreAudioTypes-11.0.0/Library/Frameworks (framework directory)
 /nix/store/y38j642asaqci7mry5hz0yfyrga1ahn4-apple-framework-IOKit-11.0.0/Library/Frameworks (framework directory)
 /nix/store/1r9344446jjd34kr7wl8ijjnswir4vmc-apple-framework-DiskArbitration-11.0.0/Library/Frameworks (framework directory)
 /nix/store/9i9k3xqbb6j4w7ir6hbn66ddyh3bsl2a-apple-framework-NetFS-11.0.0/Library/Frameworks (framework directory)
 /nix/store/mf7qx60b8hc6mvpm242ldw09jhpy5xg5-apple-framework-Security-11.0.0/Library/Frameworks (framework directory)
 /nix/store/i90055z39g08n2jip1zvagbrvc0s7n49-apple-framework-ServiceManagement-11.0.0/Library/Frameworks (framework directory)
 /nix/store/lc82lgkvqlvy11sgkr9ci2fp25pymq2r-apple-framework-SystemConfiguration-11.0.0/Library/Frameworks (framework directory)
 /nix/store/82m40ksmhhjhcmbfc8n8vip58wbq8cxm-apple-framework-CoreFoundation-11.0.0/Library/Frameworks (framework directory)
 /nix/store/22a6p8311lg3z4ksrjbj018i7z3vgkva-clang-wrapper-16.0.6/resource-root/include
 /nix/store/kk9lnicm474vif1bmhf1268xzrn7z35f-libSystem-11.0.0/include
End of search list.
In file included from otp_db.c:42:
/nix/store/kk9lnicm474vif1bmhf1268xzrn7z35f-libSystem-11.0.0/include/ndbm.h:89:3: error: typedef redefinition with different types ('struct datum' vs 'struct datum')
} datum;
  ^
/nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include/db.h:2749:3: note: previous definition is here
} datum;
  ^
In file included from otp_db.c:42:
/nix/store/kk9lnicm474vif1bmhf1268xzrn7z35f-libSystem-11.0.0/include/ndbm.h:95:3: error: typedef redefinition with different types ('struct DBM' vs 'struct __db')
} DBM;
  ^
/nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include/db.h:2729:21: note: previous definition is here
typedef struct __db DBM;
                    ^
In file included from otp_db.c:42:
/nix/store/kk9lnicm474vif1bmhf1268xzrn7z35f-libSystem-11.0.0/include/ndbm.h:116:6: error: conflicting types for '__db_ndbm_open'
DBM     *dbm_open(const char *, int, mode_t);
         ^
/nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include/db.h:2764:27: note: expanded from macro 'dbm_open'
#define dbm_open(a, b, c)       __db_ndbm_open(a, b, c)
                                ^
/nix/store/9k9jnxir5avnmv5f4nqvqr8g6mhnbjqr-db-5.3.28-dev/include/db.h:3090:6: note: previous declaration is here
DBM     *__db_ndbm_open __P((const char *, int, int));
         ^
3 errors generated.
make[2]: *** [Makefile:739: otp_db.lo] Error 1
make[2]: Leaving directory '/private/tmp/tmp.LIUy4MZguk/source/lib/otp'
make[1]: *** [Makefile:590: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/tmp.LIUy4MZguk/source/lib'
make: *** [Makefile:634: all-recursive] Error 1
```

It looks like both `libSystem` and `db` come with similar headers, but still slightly different (e.g. one uses `int` and another `size_t` for the same fields.), so the compiler fails.

----

This doesn't resolve all the issues. I also get the following:

```
Making check in roken
make[2]: Entering directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-0/source/lib/roken'
make  check-am
make[3]: Entering directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-0/source/lib/roken'
make  base32-test base64-test getaddrinfo-test getifaddrs-test hex-test test-auxval test-getuserinfo test-readenv resolve-test parse_bytes-test parse_reply-test parse_time-test snprintf-test strpftime-test timeval tsearch-test
make[4]: Entering directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-0/source/lib/roken'
  CC       base32-test.o
  CCLD     base32-test
  CC       base64-test.o
  CCLD     base64-test
  CC       getaddrinfo-test.o
  CCLD     getaddrinfo-test
  CC       getifaddrs-test.o
  CCLD     getifaddrs-test
  CC       hex-test.o
  CCLD     hex-test
make[4]: 'test-auxval' is up to date.
  CC       test-getuserinfo.o
  CCLD     test-getuserinfo
  CC       test-readenv.o
  CC       test-mem.o
  CCLD     test-readenv
make[4]: 'resolve-test' is up to date.
  CC       parse_bytes-test.o
  CCLD     parse_bytes-test
  CC       parse_reply_test-parse_reply-test.o
  CC       parse_reply_test-resolve.o
  CCLD     parse_reply-test
Undefined symbols for architecture arm64:
  "_res_9_dn_expand", referenced from:
      _parse_reply in parse_reply_test-resolve.o
      _parse_record in parse_reply_test-resolve.o
  "_res_9_search", referenced from:
      _rk_dns_lookup in parse_reply_test-resolve.o
ld: symbol(s) not found for architecture arm64
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [Makefile:1352: parse_reply-test] Error 1
make[4]: Leaving directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-0/source/lib/roken'
make[3]: *** [Makefile:2674: check-am] Error 2
make[3]: Leaving directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-0/source/lib/roken'
make[2]: *** [Makefile:2677: check] Error 2
make[2]: Leaving directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-0/source/lib/roken'
make[1]: *** [Makefile:590: check-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-heimdal-7.8.0-unstable-2024-09-10.drv-0/source/lib'
make: *** [Makefile:634: check-recursive] Error 1
```

So I also have to add libresolv as a dependency for darwin. (I am not sure if it should be `nativeBuildInputs` or `buildInputs` - it seems like maybe it's needed for `check` phase only?)

---

I have a fix for these, I'll send a PR in a sec. I am not sure it's acceptable for Linux as-is. I hope CI will be able to confirm.

--- a/pkgs/development/libraries/kerberos/heimdal.nix
+++ b/pkgs/development/libraries/kerberos/heimdal.nix
@@ -117,17 +117,15 @@ stdenv.mkDerivation {
   # (check-ldap) the bdb backend got deprecated in favour of mdb in openldap 2.5.0,
   #              but the heimdal tests still seem to expect bdb as the openldap backend.
   #              This might be fixed upstream in a future update.
-  patchPhase = ''
-    runHook prePatch
-
+  postPatch = ''
     substituteInPlace tests/ldap/slapd-init.in \
-      --replace 'SCHEMA_PATHS="' 'SCHEMA_PATHS="${openldap}/etc/schema '
+      --replace-fail 'SCHEMA_PATHS="' 'SCHEMA_PATHS="${openldap}/etc/schema '
     substituteInPlace tests/ldap/check-ldap.in \
-      --replace 'PATH=' 'PATH=${openldap}/libexec:${openldap}/bin:'
+      --replace-fail 'PATH=' 'PATH=${openldap}/libexec:${openldap}/bin:'
     substituteInPlace tests/ldap/slapd.conf \
-      --replace 'database	bdb' 'database mdb'
-
-    runHook postPatch
+      --replace-fail 'database	bdb' 'database mdb'
+    substituteInPlace tests/kdc/check-iprop.in \
+      --replace-fail '/bin/pwd' 'pwd'
   '';
 
   # (test_cc) heimdal uses librokens implementation of `secure_getenv` on darwin,

